### PR TITLE
boards: Add support for B-U585I-IOT02A

### PR DESCRIPTION
**Pull request by erwango**
This is only basic board support

### CODE REVIEW

File: boards/arm/b_u585i_iot02a/Kconfig.defconfig [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# B_L4S5I_IOT01A discovery kit board configuration
```
Comment (FRASTM) [inline boards/arm/b_u585i_iot02a/Kconfig.defconfig:1]: ` # B_U585I_IOT02A discovery kit `

File: boards/arm/b_u585i_iot02a/doc/index.rst [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+B_U585I_IOT02A Discovery kit has 9 GPIO controllers (from A to I). These controllers are responsible for pin muxing,
+input/output, pull-up, etc.
+
+For mode details please refer to `B U585I IOT02A board User Manual`_.
+
+Default Zephyr Peripheral Mapping:
+----------------------------------
+
+- UART_1 TX/RX : PA9/PA10 (ST-Link Virtual Port Com)
+- LPUART_1 TX/RX : PA0/PA1 (Arduino Serial)
```
Comment (FRASTM) [inline boards/arm/b_u585i_iot02a/doc/index.rst:197]: no but PC0/PC1

File: boards/arm/b_u585i_iot02a/doc/index.rst [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+- LD1 : PA5
```
Comment (FRASTM) [inline boards/arm/b_u585i_iot02a/doc/index.rst:198]: wrong  pins according to the DTS

**Review by FRASTM — APPROVED** [review]

**Review by jeromecoutant — APPROVED** [review]
Code review OK

**Review by galak — APPROVED** [review]

File: boards/arm/b_u585i_iot02a/doc/index.rst [hunk 4]
```diff
@@ -0,0 +1,301 @@
...
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/basic/blinky
+   :board: nucleo_u575zi_q
+   :gen-args: -r openocd
+   :goals: debug
+
+
+.. _B U585I IOT02A Discovery kit website:
+   https://www.st.com/en/evaluation-tools/bb-u585i-iot02a.html
```
Comment (fpistm) [inline boards/arm/b_u585i_iot02a/doc/index.rst:286]:
```suggestion
   https://www.st.com/en/evaluation-tools/b-u585i-iot02a.html
```

File: soc/arm/st_stm32/stm32u5/Kconfig.defconfig.stm32u585xx [hunk 5]
```diff
@@ -0,0 +1,14 @@
+# ST Microelectronics STM32U585XX MCU
```
Comment (fpistm) [inline soc/arm/st_stm32/stm32u5/Kconfig.defconfig.stm32u585xx:1]:
```suggestion
# STMicroelectronics STM32U585XX MCU
```

**Review by fpistm — CHANGES_REQUESTED** [review]

File: soc/arm/st_stm32/stm32u5/Kconfig.defconfig.stm32u585xx (hunk 5 above)
Comment (erwango) [reply soc/arm/st_stm32/stm32u5/Kconfig.defconfig.stm32u585xx]: Fixed, txs @fpistm

**Review by fpistm — APPROVED** [review]
LGTM

**Pull request merged**